patterns/1-initial/modular-code.md

```diff
@@ -8,54 +8,66 @@ The lack of modularization in the software architecture prevents reuseability, a
-    - Modular projects enable team members to focus on their own smaller assigned tasks.
-    - Modular code makes Agile implementation easier.
-    - Modular efforts increase autonomy and accountability for Engineers, which can raise satisfaction and quality.
-* Select certain "success projects," teams that will develop reusable code and demonstrate the long term success. This can help motivate others (they see what is possible and what is in it for them). Transparency is critical.
-* Offer education. Modular code is well-understood; there is a lot of literature in favor of this.
-* Acknowledge and drive acceptance for the cost of modularization. Build time into the release schedule for this.
-* Companies moving to use more open source code will appreciate modularity more over time.
-* Mitigate risk and fear of quality degradation from accepting InnerSource contributions.
-* Establish a checklist of elements to be checked off to classify a component as reusable.
-    - There could be requirements on tests, tools and documentation before considering a component as reusable
-    - Introduce refactoring as organic effort and strive for it to become part of the culture. Refactoring leads to more modular code and reduction of technical debt.
-* Establish standards on testing methodology, labeling of repos.
+Modular code is far more reusable. Decoupling functionality into smaller narrow-focused modules eases
+and therefore promotes external contributions and onboarding of newcomers.
+
+* Offer education.
+  * On why: Lack of modularity is technical debt. It increases costs but hides them by shifting these into the future.
+  * On how: Modular code is well-understood; there is a lot of literature in favor of this.
+  * Select certain "success projects", teams that will develop reusable code and demonstrate the long term success. This can help motivate others (they see what is possible and what is in it for them). Transparency is critical.
+* Introduce refactoring as organic effort and strive for it to become part of the culture.
+  * Acknowledge and drive acceptance for the investion of modularization. Build time into the release schedule for this.
+* Search for modularizable functionalities:
+  * Identify common functionalities and code blocks with potential for profitable modularity.
+  * Establish a checklist of elements to be checked off to classify a component as reusable.
+    There could be requirements on tests, tools and documentation before considering a component as reusable.
+    Establish standards on testing methodology, labeling of repos.
+* Eat the elephant in small bites. Break modularization down into baby steps, like
+  1. find internal would-be customers. Better ask them over guessing on your own.
+  1. define decoupling scope and prefferred decoupling methods,
+  1. develop automated testing protection
+  1. document usage of modules.
 
```

Comment:
I spoke to @fioddor about this and I wish to add one more line here which I think is super useful
```suggestion
* More information and techniques are available from the book Working Effectively with Legacy Code by Michael Feathers and Refactoring: Improving the Design of Existing Code by Martin Fowler

```